File: src/test/ImageButton.java
package test;


import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.input.MouseEvent;

public class ImageButton extends ImageView {

    Image up, down;


    public ImageButton(String upImage, String downImage) {
        super(upImage);

        up = new Image(upImage);
        down = new Image(downImage);

        setOnMousePressed((MouseEvent e) -> {
            setImage(down);
        });

        setOnMouseReleased((MouseEvent e) -> {
            setImage(up);
        });
    }
}

File: src/test/ImageButton.kt
package test

import javafx.scene.image.Image
import javafx.scene.image.ImageView
import javafx.scene.input.MouseEvent

class ImageButton(upImage: String, downImage: String) : ImageView() {

    init {
        val up = Image(upImage, 128.0, 128.0, true, true)
        val down = Image(downImage, 128.0, 128.0, true, true)

        image = up

        setOnMousePressed { e: MouseEvent -> image = down }

        setOnMouseReleased { e: MouseEvent -> image = up }
    }
}
